.github/workflows/image.yml

```diff
@@ -4,9 +4,19 @@ on:
     branches: [master]
 jobs:
   build:
-    outputs:
-      image_tag: ${{ steps.get_image_tag.outputs.image_tag }}
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        arch: [amd64, arm64]
+    runs-on: |-
+      ${{fromJson('{
+        "amd64": "ubuntu-22.04",
+        "arm64": "ubuntu-22.04-arm"
+      }')[matrix.arch] }}
```

Comment:
🤯

Reply:
if you want to see some real crazy usage of fromJson, check out what cargo-dist does. it seems that one can generate matrices programmatically, put them into job output and then reference them here.

Reply:
Just to note, this is absolutely not necessary here, I did it this way to not have the runner show up in the generated job name in CI.

Reply:
> Just to note, this is absolutely not necessary here, I did it this way to not have the runner show up in the generated job name in CI.

No no, it's useful. We need to use the correct GHA runner with the corresponding arch name (amd64/arm64)

Reply:
> No no, it's useful.

It's not, you can just put it in the matrix. The purpose of this is simply to not have the runner name in the job name.

Reply:
> > No no, it's useful.
> 
> 
> 
> It's not, you can just put it in the matrix. The purpose of this is simply to not have the runner name in the job name.

Now it makes sense lol.